src/cargo/core/source.rs

```diff
@@ -87,6 +87,8 @@ enum Kind {
     Registry,
     /// represents a local filesystem-based registry
     LocalRegistry,
+    /// represent an alternate registry
+    AlternateRegistry(String),
```

Comment:
I wonder, could this be folded into the `Registry` type above? We could then either store `Option<String>` or I'd be fine having a global notion of "default registry" which is defaulted to something like `crates-io`

Reply:
It prints out a slightly different message, indicating that its an alternative registry (e.g. when you update the index). But that could also be covered by `Option<String>`.